Xplat/src/main/java/vazkii/botania/common/block/flower/functional/LooniumBlockEntity.java

```diff
@@ -78,116 +92,119 @@ public void tickFlower() {
+			return;
+		}
 
-			ItemStack stack = pickRandomLoot(world, rand);
-			if (stack.isEmpty()) {
-				return;
-			}
+		var configData = BotaniaAPI.instance().getConfigData();
+		var structureConfigs = determineStructureConfigs(configData, detectedStructures);
+		var lootTables = determineLootTables(world, detectedStructures.keySet());
 
-			int bound = RANGE * 2 + 1;
-			int xp = getEffectivePos().getX() - RANGE + rand.nextInt(bound);
-			int yp = getEffectivePos().getY();
-			int zp = getEffectivePos().getZ() - RANGE + rand.nextInt(bound);
+		if (lootTables.isEmpty()) {
+			return;
+		}
 
-			BlockPos pos = new BlockPos(xp, yp - 1, zp);
-			do {
-				pos = pos.above();
-				if (pos.getY() >= world.getMaxBuildHeight()) {
-					return;
-				}
-			} while (world.getBlockState(pos).isSuffocating(world, pos));
-			pos = pos.above();
+		var randomPick = lootTables.get(world.random.nextInt(lootTables.size()));
+		var pickedConfig = structureConfigs.getOrDefault(randomPick.key(),
+				structureConfigs.get(LooniumStructureConfiguration.DEFAULT_CONFIG_ID));
+		var pickedLootTable = randomPick.value();
 
-			double x = pos.getX() + Math.random();
-			double y = pos.getY() + Math.random();
-			double z = pos.getZ() + Math.random();
+		if (getMana() < pickedConfig.manaCost) {
+			return;
+		}
+
+		var pickedMobType = pickedConfig.spawnedMobs.getRandom(world.random).orElse(null);
+		if (pickedMobType == null) {
+			return;
+		}
 
-			// TODO: Mob types and weights should be defined per structure
-			Mob entity = spawnMob(world);
-			if (entity == null) {
+		ItemStack stack = pickRandomLootItem(world, pickedLootTable);
+		if (stack.isEmpty()) {
+			return;
+		}
+
+		int bound = RANGE * 2 + 1;
+		int xp = getEffectivePos().getX() - RANGE + world.random.nextInt(bound);
+		int yp = getEffectivePos().getY();
+		int zp = getEffectivePos().getZ() - RANGE + world.random.nextInt(bound);
+
+		BlockPos pos = new BlockPos(xp, yp - 1, zp);
+		do {
+			pos = pos.above();
+			if (pos.getY() >= world.getMaxBuildHeight()) {
 				return;
 			}
+		} while (world.getBlockState(pos).isSuffocating(world, pos));
+		pos = pos.above();
 
-			entity.absMoveTo(x, y, z, world.random.nextFloat() * 360F, 0);
-			entity.setDeltaMovement(Vec3.ZERO);
+		double x = pos.getX() + Math.random();
+		double y = pos.getY() + Math.random();
+		double z = pos.getZ() + Math.random();
 
-			// TODO: mob attribute modifications should be defined along with mob configuration
-			entity.getAttribute(Attributes.MAX_HEALTH).addPermanentModifier(new AttributeModifier("Loonium Modififer Health", 2, AttributeModifier.Operation.MULTIPLY_BASE));
-			entity.setHealth(entity.getMaxHealth());
-			entity.getAttribute(Attributes.ATTACK_DAMAGE).addPermanentModifier(new AttributeModifier("Loonium Modififer Damage", 1.5, AttributeModifier.Operation.MULTIPLY_BASE));
+		Entity entity = pickedMobType.type.create(world);
+		if (!(entity instanceof Mob mob)) {
```

Comment:
booring :p 
let us spawn TNT :3
or maybe this breaks the attribute and mob effect stuff. and loot tables i guess..

Reply:
yeah probably not really viable